legend-engine-shared-core/src/main/java/org/finos/legend/engine/shared/core/identity/credential/LegendKerberosCredential.java

```diff
@@ -45,7 +45,11 @@ public void setSubject(Subject subject)
     @Override
     public boolean isValid()
     {
-        KerberosTicket kerberosTicket = subject.getPrivateCredentials(KerberosTicket.class).iterator().next();
-        return kerberosTicket!=null && kerberosTicket.isCurrent();
+        if (subject.getPrivateCredentials(KerberosTicket.class) != null)

```

Comment:
should subject.getPrivateCredentials(KerberosTicket.class) be assigned to a local variable to match for the check and the usage? (Or is there no chance of it being modified between these two lines?)

Reply:
done